### ISSUE TITLE
[Suggestion] Possible Additional Features

### ISSUE THREAD

**Issue by MythrilBagels** [AstralOrdana/Enchantery#19]
Enchanting is one of the several aspects of Minecraft that deserves an overhaul, and this mod is certainly up there in addressing that.
However, I have a few suggestions inspired by some other mods that have tried in the past.

### Suggestion 1 - Enchantment Slots:
In the old mod Enchanting Overhauled (which seems to be abandoned, unfortunately), there was in place a limit of 3 to the number of enchantments that could be applied to equipment. This encouraged more thought into choosing enchantments and encouraged actually creating different tools with different sets of enchantments. This also ties to the fact the Enchantment table has 3 slots for displaying Enchantments, which also ties into a later suggestion. From my experience playing with that mod, I find it to be a great addition to the enchanting system and I would certainly love to see it brought back.

### Suggestion 2 - Enchanted Books Applied Through the Enchanting Table:
Another feature that I adored from Enchanting Overhauled was moving Book application from the anvil to the Enchanting Table, which is similar to what Enchantery does with Chiseled Bookshelves, except far less overpowered since they are used up upon enchanting (Except for the special "enchanted tomes", which allowed for endless enchanting). It truly felt more magical than applying them at an anvil.

### Suggestion 3 - Upgrading Enchantments:
The last feature from Enchanting Overhauled specifically that I will suggest. There was a system in place where the Enchanting Table would only apply the first level of the selected enchantment, but let Players pay more experience to level up that enchantment, adding a really solid sense of progression. 

### Suggestion 4 - Chiseled Bookshelf Enchanting Power:
Rather than the current system within the mod, where enchanted books inside Chiseled Bookshelves can be infinitely applied to items, I propose that they instead be used as an option of increasing the Enchanting Power of the enchanting table, based on the number of books within. Alongside this, an overall increase of the maximum enchanting power to make way for how many books the chiseled bookshelves can hold, and the extra bookshelf space Enchantery makes way for (the 7x7 change). This also ties into my last suggestion as well.

### Suggestion 5 - Enchanting Power Requirements:
Applying enchantments should have a minimum Enchanting Power requirement, so you can't just place an enchantment table, and a chiseled bookshelf with a mending book, and get infinite cheap mending. It should require the Player actually increase the enchanting power of the table. This should also be configurable, for those who would want to change balancing, or integrate external modded enchantments, but that likely goes without question.

In my mind, this would add together to create the enchanting system of my dreams, and I truly hope you consider these suggestions.

### TL:DR
- 3 Enchantment Slots.
- Books applied at an Enchanting Table (But not through Chiseled Bookshelves).
- Upgrading the level of Enchantments one by one.
- Chiseled Bookshelf increasing Enchanting Power instead.
- Varying Enchanting Power requirement.
